src/Pelagos/Bundle/AppBundle/Resources/public/js/data-discovery.js

```diff
@@ -112,39 +358,42 @@ function showDatasets(by,id) {
         }
         $("#packageLink").attr("href",newlink);
     }
-    geo_filter = "";
-    if (trees["tree"].geo_filter) {
-        geo_filter = trees["tree"].geo_filter;
-    }
+
+    //get number of results for each tabs
     $.ajax({
-        "url": Routing.generate("pelagos_app_ui_datadiscovery_datasets"),
+        "url": Routing.generate("pelagos_app_ui_datadiscovery_count"),
         "data": {
-            "filter": jQuery("#filter-applied").val(),
+            "filter": filter,
             "by": by,
             "id": id,
             "geo_filter": geo_filter
         },
         "success": function(data) {
             $("#dataset_listing_wrapper .spinner").hide();
+            loadData(by, id);
             $("#dataset_listing").html(data);
             $("#tabs").tabs({
                 activate: function(event, ui) {
-                    if ($("#show_all_extents_checkbox").is(":checked")) {
-                        var selectedTab = $("#tabs").tabs("option","active");
-                        myGeoViz.removeAllFeaturesFromMap();
-                        if (datasets[selectedTab]) {
-                            for (var i=0; i<datasets[selectedTab].length; i++) {
-                                myGeoViz.addFeatureFromWKT(datasets[selectedTab][i].geom,{"udi":datasets[selectedTab][i].udi});
-                            }
-                        }
+                    var activeTabIndex = getActiveTabIndex();
+                    if (datasetList[activeTabIndex].length == 0) {
+                        loadData(by, id);
+                    }
+
+                    if ($("#show_extents_checkbox").is(":checked")) {
+                        displayActiveTabExtents();
                     }
                 }
-            }
-            );
+            });
             enableFilterButton();
             $("#clear-button").button("enable");
             $("#drawGeoFilterButton").button("enable");
-            if (myGeoViz.getFilter()) {
+            //this triggers infinite scrolling
+            $(".viewport").has(".datasets").scroll(function(){
+                if ($(this).scrollTop() + $(this).innerHeight() >= $(this)[0].scrollHeight) {
```

Comment:
scroll doesn't fire at certain resolutions, I fixed it by subtracting a from scrollHeight like this:
`$(this)[0].scrollHeight-10)`

As discussed here: https://stackoverflow.com/questions/3898130/check-if-a-user-has-scrolled-to-the-bottom

Reply:
Actually -1 seems to works just about right.